File: 3309-maximum-possible-number-by-binary-concatenation/3309-maximum-possible-number-by-binary-concatenation.cpp
class Solution {
public:
    int maxGoodNumber(vector<int>& a) {
        sort(begin(a), end(a));
        int ans = 0;
        do {
            int cur = 0;
            for (int x : a) {
                bool flag = false;
                for (int i = 10; i >= 0; --i) {
                    if (x & (1 << i)) {
                        flag = true;
                    }
                    if (flag) cur = cur*2 + ((x >> i) & 1);
                }
            }
            ans = max(ans, cur);
        } while (next_permutation(begin(a), end(a)));
        return ans;
    }
};

File: 3309-maximum-possible-number-by-binary-concatenation/3309-maximum-possible-number-by-binary-concatenation.py
import functools
class Solution:
    def maxGoodNumber(self, a: List[int]) -> int:
        
        a=[bin(x)[2:] for x in a]


        def cmp(a,b):
            if int(a+b)>int(b+a):
                return -1
            else:
                return 1
        a.sort(key=functools.cmp_to_key(cmp))
        res=""
        for x in a:
            res+=x
        return int(res,2)